### PR TITLE
Updating some places to use the new CustomUserMessageAuthenticationException

### PR DESCRIPTION
**Pull request by weaverryan**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.8+ |
| Fixed tickets | #5736 |

### CODE REVIEW

**Comment by weaverryan** [comment]
More is added in #5265 also

File: cookbook/security/custom_password_authenticator.rst [hunk 1]
```diff
@@ -47,7 +47,8 @@ the user::
             try {
                 $user = $userProvider->loadUserByUsername($token->getUsername());
             } catch (UsernameNotFoundException $e) {
-                throw new AuthenticationException('Invalid username or password');
+                // error will be shown to the client
```
Comment (WouterJ) [inline cookbook/security/custom_password_authenticator.rst]: error -> message ?

**Comment by WouterJ** [comment]
:+1:

**Comment by xabbuh** [comment]
I think we should add a caution directive here to not pass anything blindly here which may leak security related information (something like what is [done in the class docblock](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/Exception/CustomUserMessageAuthenticationException.php#L17-L19)).

**Comment by weaverryan** [comment]
Thanks guys! I added some inline cautions :)

**Pull request merged**